MultitagInspector.cs

```diff
@@ -97,11 +97,12 @@ public override void OnInspectorGUI()
 
         var forceSubmit = false;
         var forceClear = false;
-        var curKeyboardInput = Event.current;
+        var curKeyboardInput = Event.current; 
         if (curKeyboardInput.isKey)
         {
-            if (curKeyboardInput.keyCode == KeyCode.KeypadEnter
+            if ((curKeyboardInput.keyCode == KeyCode.KeypadEnter
                 || curKeyboardInput.keyCode == KeyCode.Return)
+                && curKeyboardInput.type == EventType.KeyUp)
```

Comment:
the UX of having to release the key for it to take is bad. you should check if the current input field is empty and only filter by `KeyUp` then.

Reply:
or better yet, store a flag along the lines of `justSubmitted` that gets set to true when you press enter. then wait for `keyUp` to reset it. then you keep it responsive and consistent in both cases.